sources/packages/web/src/components/institutions/modals/AddInstitutionUserModal.vue

```diff
@@ -1,245 +1,172 @@
-          toast.success(
-            "User Added",
-            `${selectUser.value.name} Successfully Added!`,
-          );
-        } catch (excp) {
-          const errorMessage =
-            excp === LEGAL_SIGNING_AUTHORITY_EXIST
-              ? LEGAL_SIGNING_AUTHORITY_MSG
-              : "An error happened during the update process.";
-          toast.error("Unexpected error", errorMessage);
-        }
-        closeAddUser();
-        context.emit("getAllInstitutionUsers");
-      } else {
-        if (!selectUser.value.code) {
-          invalidName.value = true;
-        }
-        if (payLoad.value?.location?.length === 0) {
-          invalidUserType.value = true;
+
+    // Watch for changes on institutionId to reload the UI.
+    watch(
+      () => props.institutionId,
+      async () => {
+        await loadLocations();
+        if (props.hasBusinessGuid) {
+          // Load BCeID users only for institutions that have a business guid.
+          await loadBCeIDBusinessUsers();
         }
+      },
+      { immediate: true },
+    );
+
+    // Creates the user and closes the modal.
+    const submit = async () => {
+      const form = institutionUserManagement.value.userForm as VForm;
+      const validationResult = await form.validate();
+      if (!validationResult.valid) {
+        return;
       }
+      const userManagementModel = institutionUserManagement.value
+        .formModel as UserManagementModel;
+      try {
+        await InstitutionService.shared.createInstitutionUserWithAuth(
+          userManagementModel.selectedBCeIDUser,
+          userManagementModel.isAdmin,
+          userManagementModel.isLegalSigningAuthority,
+          userManagementModel.locationAuthorizations,
+        );
+        toast.success("User created", "User successfully created.");
+        resolvePromise(true);
+      } catch {
+        toast.error(
+          "Unexpected error",
+          "An error happened while creating the user.",
+        );
+      }
+    };
+
+    // Closed the modal dialog.
+    const cancel = () => {
+      resolvePromise(false);
     };
 
-    onMounted(async () => {
-      // call institution location
-      await getInstitutionLocationList();
-      //  Get All users from Bceid;
-      const bceidUsers = await UserService.shared.getBCeIDAccounts();
-      usersList.value = bceidUsers
-        ? bceidUsers?.accounts.map((el: any) => ({
-            name: el.displayName,
-            code: el.userId,
-            id: el.guid,
-          }))
-        : [];
-    });
     return {
-      usersList,
-      selectUser,
-      institutionLocationList,
-      closeAddUser,
-      submitAddUser,
-      isAdmin,
-      invalidName,
-      invalidUserType,
-      display,
-      selectedAdminRole,
+      showDialog,
+      showModal,
+      submit,
+      cancel,
+      initialData,
+      bceidUsers,
+      institutionUserManagement,
     };
   },
 };
 </script>
+<style scoped>
+.bceid-input {
+  width: 300px;
```

Comment:
just wanted to check, since its in `px` will it break in smaller resolutions?

Reply:
Apparently, there is something wrong with the inputs in the current Vuetify version, which is why the other similar style was created for the search input texts. I can try to update to beta 5 in the next PR and see how it goes. It is supposed to be a temporary solution. Maybe I should add a comment.